src/types.ts

```diff
@@ -1,21 +1,40 @@
-import {
-  Context,
-  Fluent,
-  FluentBundleOptions,
-  FluentOptions,
-  FluentVariable,
-  LocaleId,
-  TranslationContext,
-} from "./deps.ts";
+import { Context, FluentBundle, FluentVariable } from "./deps.ts";
+import { Fluent } from "./fluent.ts";
+import { WarningHandler } from "./warning.ts";
+
+export type MaybeArray<T> = T | T[];
+export type LocaleId = string;
+export type FluentBundleOptions = ConstructorParameters<typeof FluentBundle>[1];
+
+export type FilepathOrSource = { filePath: string } | { source: string };
+
+export type AddTranslationOptions = {
+  locales: MaybeArray<string>;
+  bundleOptions?: FluentBundleOptions;
+  isDefault?: boolean;
+} & FilepathOrSource;
+
+export type LoadLocaleOptions = FilepathOrSource & {
+  isDefault?: boolean;
+  bundleOptions?: FluentBundleOptions;
+};
+
+export interface FluentOptions {
+  warningHandler?: WarningHandler;
+}
 
 export type LocaleNegotiator<C extends Context = Context> = (ctx: C) =>
   | LocaleId
   | undefined
   | PromiseLike<LocaleId | undefined>;
 
-export type TranslateFunction = (
+export type TranslationVariables<K extends string = string> = Record<
+  K,
+  FluentVariable
+>;
+export type TranslateFunction = <K extends string>(
   key: string,
-  context?: TranslationContext,
+  context?: TranslationVariables<K>,
```

Comment:
changing the name of the type should also change the name of the variable. Guess this change is relevant in other places too. I think I suggested all of them but better check another time.
```suggestion
  variables?: TranslationVariables<K>,
```